Chevrotain fails to compile with Typescript and noImplicitAny

Here are the errors:

```
node_modules/chevrotain/lib/chevrotain.d.ts(698,42): error TS7019: Rest parameter 'args' implicitly has an 'any[]' type.
node_modules/chevrotain/lib/chevrotain.d.ts(698,67): error TS7006: Parameter 'T' implicitly has an 'any' type.
node_modules/chevrotain/lib/chevrotain.d.ts(751,39): error TS7006: Parameter 'number' implicitly has an 'any' type.
node_modules/chevrotain/lib/chevrotain.d.ts(770,40): error TS7006: Parameter 'number' implicitly has an 'any' type.
node_modules/chevrotain/lib/chevrotain.d.ts(774,40): error TS7006: Parameter 'number' implicitly has an 'any' type.
node_modules/chevrotain/lib/chevrotain.d.ts(778,40): error TS7006: Parameter 'number' implicitly has an 'any' type.
node_modules/chevrotain/lib/chevrotain.d.ts(782,40): error TS7006: Parameter 'number' implicitly has an 'any' type.
node_modules/chevrotain/lib/chevrotain.d.ts(786,40): error TS7006: Parameter 'number' implicitly has an 'any' type.
node_modules/chevrotain/lib/chevrotain.d.ts(1123,47): error TS7006: Parameter 'number' implicitly has an 'any' type.
```
I fixed them in my local copy of chevrotain.d.ts but I don't see how/where you are generating that file here.

Here's my fixed version: https://gist.github.com/ibdknox/47efe725ea4f99cafac90dbe34b6e4ae

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,13 @@
+## X.Y.Z (INSERT_DATE_HERE)
+
+#### Minor Changes
+- [Upgrade to TypeScript 2.3.1](https://github.com/SAP/chevrotain/pull/458)
+
+#### Documentation
+- [Ensure chevrotain.d.ts is compatiable with noImplicitAny.](https://github.com/SAP/chevrotain/issues/459)
+
+
+
 ## 0.28.1 (4-18-2017)
 
 #### Minor Changes

--- a/package.json
+++ b/package.json
@@ -48,8 +48,8 @@
     "release_minor": "git fetch && git rebase && node scripts/pre_release_build.js minor && grunt build && node scripts/post_release_build.js minor && node ../chevrotain_docs/scripts/upload_docs.js",
     "unit_tests": "node_modules/.bin/_mocha ./lib/test/**/*spec.js",
     "compile": "tsc",
-    "compile_definitions": "tsc test_integration/definitions/es6_modules.ts --outDir dev/garbage",
-    "compile_definitions_namespace": "tsc test_integration/definitions/namespaces.ts --module none --outDir dev/garbage",
+    "compile_definitions": "tsc --noImplicitAny test_integration/definitions/es6_modules.ts --outDir dev/garbage",
+    "compile_definitions_namespace": "tsc --noImplicitAny test_integration/definitions/namespaces.ts --module none --outDir dev/garbage",
     "lint": "tslint --project tsconfig.json"
   },
   "dependencies": {},

--- a/src/parse/parser_public.ts
+++ b/src/parse/parser_public.ts
@@ -731,7 +731,7 @@ export class Parser {
      *
      * @return {Function():boolean} a lookahead function that will try to parse the given grammarRule and will return true if succeed.
      */
-    protected BACKTRACK<T>(grammarRule:(...args) => T, isValid:(T) => boolean):() => boolean {
+    protected BACKTRACK<T>(grammarRule:(...args:any[]) => T, isValid:(T:any) => boolean):() => boolean {
         return function () {
             // save org state
             this.isBackTrackingStack.push(1)
@@ -838,7 +838,7 @@ export class Parser {
      * Convenience method equivalent to SUBRULE1
      * @see SUBRULE1
      */
-    protected SUBRULE<T>(ruleToCall:(number) => T, args:any[] = undefined):T {
+    protected SUBRULE<T>(ruleToCall:(idx:number) => T, args:any[] = undefined):T {
         return this.subruleInternal(ruleToCall, 1, args)
     }
 
@@ -860,35 +860,35 @@ export class Parser {
      * @param {*[]} args - The arguments to pass to the invoked subrule.
      * @returns {*} - The result of invoking ruleToCall.
      */
-    protected SUBRULE1<T>(ruleToCall:(number) => T, args:any[] = undefined):T {
+    protected SUBRULE1<T>(ruleToCall:(idx:number) => T, args:any[] = undefined):T {
         return this.subruleInternal(ruleToCall, 1, args)
     }
 
     /**
      * @see SUBRULE1
      */
-    protected SUBRULE2<T>(ruleToCall:(number) => T, args:any[] = undefined):T {
+    protected SUBRULE2<T>(ruleToCall:(idx:number) => T, args:any[] = undefined):T {
         return this.subruleInternal(ruleToCall, 2, args)
     }
 
     /**
      * @see SUBRULE1
      */
-    protected SUBRULE3<T>(ruleToCall:(number) => T, args:any[] = undefined):T {
+    protected SUBRULE3<T>(ruleToCall:(idx:number) => T, args:any[] = undefined):T {
         return this.subruleInternal(ruleToCall, 3, args)
     }
 
     /**
      * @see SUBRULE1
      */
-    protected SUBRULE4<T>(ruleToCall:(number) => T, args:any[] = undefined):T {
+    protected SUBRULE4<T>(ruleToCall:(idx:number) => T, args:any[] = undefined):T {
         return this.subruleInternal(ruleToCall, 4, args)
     }
 
     /**
      * @see SUBRULE1
      */
-    protected SUBRULE5<T>(ruleToCall:(number) => T, args:any[] = undefined):T {
+    protected SUBRULE5<T>(ruleToCall:(idx:number) => T, args:any[] = undefined):T {
         return this.subruleInternal(ruleToCall, 5, args)
     }
 
@@ -1462,7 +1462,7 @@ export class Parser {
         return nextPossibleTokenTypes
     }
 
-    protected subruleInternal<T>(ruleToCall:(number) => T, idx, args:any[]) {
+    protected subruleInternal<T>(ruleToCall:(idx:number) => T, idx:number, args:any[]) {
         let ruleResult = ruleToCall.call(this, idx, args)
         this.cstPostNonTerminal(ruleResult, (<any>ruleToCall).ruleName)
 

